src/main/scala/tech/sourced/gemini/ReportSparkApp.scala

```diff
@@ -1,34 +1,67 @@
 package tech.sourced.gemini
 
+import com.datastax.driver.core.Cluster
+
 object ReportSparkApp extends App {
   def printUsage(): Unit = {
-    println("Usage: ./report <repository>")
+    println("Usage: ./report [--use-group-by|--condensed]")
     println("")
-    println("Finds duplicated files among hashed repositories")
-    println("  <repository> - repository url, example: github.com/src-d/go-git.git")
+    println("Finds duplicate files among hashed repositories." +
+      "\n\tIt uses as many queries as distinct files are stored in the database"
+    )
+    println("  --use-group-by" +
+      "\n\tIt uses as many queries as unique duplicate files are found, plus one. (only for Apache Cassandra database)"
+    )
+    println("  --condensed displays only the duplicate blob_hash and the number of occurrences." +
+      "\n\tIt uses only one query to find the duplicates. (only for Apache Cassandra database)"
+    )
     System.exit(2)
   }
 
-  if (args.length <= 0) {
+  def print(report: Report): Unit = {
+    report match {
+      case e if e.empty() => println(s"No duplicates found.")
+      case r: ReportGrouped => println(s"Duplicates found:\n\t" + (r.v mkString "\n\t"))
```

Comment:
I'm only learning Scala, but I saw class matching for such cases very often in examples.
I mean
```
case ReportGrouped(v)=> print(v)
```
JFYI.

Reply:
many thanks for the advice!